### PR TITLE
MID: skip digits produced before the beginning of the TF

### PR DESCRIPTION
**Pull request by dstocco**
This should address the JIRA ticket: https://its.cern.ch/jira/browse/O2-5395.

### CODE REVIEW

File: Steer/DigitizerWorkflow/src/MIDDigitizerSpec.cxx [hunk 1]
```diff
@@ -93,6 +96,11 @@ class MIDDPLDigitizerTask : public o2::base::BaseDPLDigitizer
     for (int collID = 0; collID < irecords.size(); ++collID) {
       // for each collision, loop over the constituents event and source IDs
       // (background signal merging is basically taking place here)
+
+      // Skip digits produced before the first orbit
+      if (irecords[collID].orbit < firstTFOrbit) {
```
Comment (sawenzel) [inline Steer/DigitizerWorkflow/src/MIDDigitizerSpec.cxx]: strictly speaking this seems to be skipping collisions happening before the firstTFOrbit.

Is this fully correct for MID? Could it not happen that there are digits created after the first orbit (due to time of flight) ... originating from collisions happening before?

Comment (sawenzel) [reply Steer/DigitizerWorkflow/src/MIDDigitizerSpec.cxx]: And what happens if the orbit is the same but the bunch crossing of the collision earlier than `FirstSampleTFIR`?

**Comment by shahor02** [comment]
The TF, hence `getFirstSampledTFIR()` always starts with bc=0.

File: Steer/DigitizerWorkflow/src/MIDDigitizerSpec.cxx (hunk 1 above)
Comment (dstocco) [reply Steer/DigitizerWorkflow/src/MIDDigitizerSpec.cxx]: @sawenzel : the MID time resolution is better than 25 ns, so we do do not have a dispersion like e.g. the MCH. Also, when a new TF starts, the UL waits for the answer to the HB trigger from all of the boards in the electronics before sending the data.
However, we do have a delay of 90 BCs. This is properly accounted for in the Encoder, and corrected in the Decoder. I can include the protection also at this level if needed.

**Comment by dstocco** [comment]
But usually in the filling scheme the first BCs are not filled. Is this taken into account in simulations? Otherwise I will be systematically rejecting BCs in simulations that I would not in data...

File: Steer/DigitizerWorkflow/src/MIDDigitizerSpec.cxx (hunk 1 above)
Comment (sawenzel) [reply Steer/DigitizerWorkflow/src/MIDDigitizerSpec.cxx]: Ok understood. Just wanted to make sure. Thanks for this work !

**Review by sawenzel — APPROVED** [review]

**Comment by Barthelemy** [comment]
@sawenzel I see that all the PRs are failing due to an issue with MID. Could it have been caused by this PR ?

**Comment by shahor02** [comment]
@Barthelemy let's see if https://github.com/AliceO2Group/AliceO2/pull/13615 fixes the problem.

**Comment by dstocco** [comment]
@shahor02 : not sure to understand. The modification that you introduce in #13615 is already present in this PR...

**Comment by dstocco** [comment]
Ok, I see now: the library dependency was not inserted in the digitizer-workflow executable. I did not get any error on Mac OS, so I did not notice. Sorry about that

**Comment by shahor02** [comment]
The dependency was added for the build with upgrades enabled but not w/o it. Apparently different CIs use different options.

**Comment by dstocco** [comment]
But I see that there is a global if: 
```
if (ENABLE_UPGRADES)
```
and then specific ifs to include the needed packages in case of upgrades:
```
$<$<BOOL:${ENABLE_UPGRADES}>:O2::ITS3Simulation>
                                        $<$<BOOL:${ENABLE_UPGRADES}>:O2::ITS3Workflow>
                                        $<$<BOOL:${ENABLE_UPGRADES}>:O2::ITS3Align>)
```
To simplify and avoid confusion, can't we simply remove the global if?

**Comment by shahor02** [comment]
In principle, yes. @f3sch , was there a reason to keep `if (ENABLE_UPGRADES)` after adding `$<$<BOOL:${ENABLE_UPGRADES}>...` ?

**Comment by f3sch** [comment]
no reason, this was overlooked and should be removed as @dstocco suggests.

**Pull request merged**